### PR TITLE
[POAE7-2181] Fix string error in batch checker and add tests for IS DISTINCT/IS NOT DISTINCT

### PR DESCRIPTION
**Pull request by Deegue**
### What changes were proposed in this pull request?
Remove empty string value initialization in constructor of batch checker.
Add string and mixed test cases and improve comments.

### Why are the changes needed?
Fix & Improvement.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
UTs.

### CODE REVIEW

**Review by winningsix — DISMISSED** [review]
Seems adding test for " IS DISTINCT/IS NOT DISTINCT" and fix string error are irrelevant? If so, we should create two PRs next time.

File: cider/util/sqldefs.h [hunk 1]
```diff
@@ -26,7 +26,8 @@
 // in sync.
 enum SQLOps {
   kEQ = 0,
-  // BitWise Equal <=> a = b or (a is null and b is null)
+  // The only difference between EQ and BW_EQ is that BW_EQ will check nullability of both
```
Comment (jikunshang) [inline cider/util/sqldefs.h:29]: `check nullability of both sides` means `(a is null and b is null)` ?

**Review by jikunshang — APPROVED** [review]

File: cider/util/sqldefs.h (hunk 1 above)
Comment (yma11) [reply cider/util/sqldefs.h:29]: Seems previous comments is more clear. They should both check nullability but for `kEQ`, it doesn't return true if both sides are null?

**Comment by Deegue** [comment]
> Seems adding test for " IS DISTINCT/IS NOT DISTINCT" and fix string error are irrelevant? If so, we should create two PRs next time.

Yes, this test exposed the bug. I will separate them next time.

File: cider/util/sqldefs.h (hunk 1 above)
Comment (Deegue) [reply cider/util/sqldefs.h:29]: See [check-equal-bitwise](https://iq.opengenus.org/check-equal-bitwise/) for more information.

**Review by jikunshang — APPROVED** [review]

**Review by yma11 — APPROVED** [review]

**Pull request merged**